pilot/pkg/model/service.go

```diff
@@ -319,19 +267,19 @@ func GetLocalityOrDefault(label, defaultLocality string) string {
 // IstioEndpoint has the information about a single address+port for a specific
 // service and shard.
 //
-// TODO: Replace NetworkEndpoint and ServiceInstance with Istio endpoints
+// TODO: Investigate removing ServiceInstance entirely.
 // - ServicePortName replaces ServicePort, since port number and protocol may not
 // be available when endpoint callbacks are made.
-// - It no longer splits into one ServiceInstance and one NetworkEndpoint - both
-// are in a single struct
 // - doesn't have a pointer to Service - the full Service object may not be available at
 // the time the endpoint is received. The service name is used as a key and used to reconcile.
 // - it has a cached EnvoyEndpoint object - to avoid re-allocating it for each request and
 // client.
 type IstioEndpoint struct {
+	// Hostname of the service. Useful if the user needs to look up the Service from the endpoint.
+	Hostname host.Name
```

Comment:
?? The PR is for renaming. If you want to tweak the model, let us do it in a separate PR.

Reply:
It's not just renaming, right? We have all these structures that are subtly different for no real benefit. So to reconcile these it requires more than just renaming

Reply:
Is this for future use?

Reply:
It's not currently used, but it is logically needed in order to be afford callers the ability to map endpoints back to services. The only mechanism the API provides for searching for services is via hostname.  The controller always has the hostname available when constructing the endpoints, so it seems a low cost addition.

Reply:
This is not right. a pod can belong to multiple services. And this assumption is all over the place. So, please nuke this and we can investigate in a separate PR. The last thing I want is this piece being assumed a given by some clueless person writing the next main feature.